conda_forge_tick/git_utils.py

```diff
@@ -614,6 +815,22 @@ def does_repository_exist(self, owner: str, repo_name: str) -> bool:
         except RepositoryNotFoundError:
             return False
 
+    @lock_git_operation()
+    def push_to_repository(
+        self, owner: str, repo_name: str, git_dir: Path, branch: str
+    ):
+        # We add the token here and remove it immediately after pushing as an additional defense-in-depth measure.
+        self.cli.add_token(git_dir, self.GIT_PLATFORM_ORIGIN, self.__token)
+
+        remote_url = self.get_remote_url(
+            owner,
+            repo_name,
+            GitConnectionMode.HTTPS,
+        )
+        self.cli.push_to_url(git_dir, remote_url, branch)
+
+        self.cli.clear_token(git_dir, self.GIT_PLATFORM_ORIGIN)
```

Comment:
This needs to use a try...finally block for safety.

Reply:
Done in 1a99cebbb5c491c95333344821ffd8a66b490a42.